src/ray/gcs/gcs_node_manager.cc

```diff
@@ -315,6 +317,126 @@ void GcsNodeManager::HandleGetAllNodeInfo(rpc::GetAllNodeInfoRequest request,
   ++counts_[CountType::GET_ALL_NODE_INFO_REQUEST];
 }
 
+// Utility function to convert GcsNodeInfo to GcsNodeInfoLight
+rpc::GcsNodeInfoLight ConvertToGcsNodeInfoLight(const rpc::GcsNodeInfo &source) {
+  rpc::GcsNodeInfoLight destination;
+  destination.set_node_id(source.node_id());
+  destination.set_node_manager_address(source.node_manager_address());
+  destination.set_node_manager_port(source.node_manager_port());
+  destination.set_object_manager_port(source.object_manager_port());
+  destination.set_state(source.state());
+  destination.mutable_death_info()->CopyFrom(source.death_info());
+  return destination;
+}
+
+void GcsNodeManager::HandleGetAllNodeInfoLight(
```

Comment:
Admittedly, this is some very copy paste code.  I'll need to look through if there's a way we can get better reuse.  Open to suggestions.